inginious/frontend/pages/admin/admin.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+
+""" Admin index page"""
+from flask import redirect, request
+from inginious.frontend.pages.utils import INGIniousAdministratorPage
+import hashlib
+
+
+class AdministrationPage(INGIniousAdministratorPage):
+    """Admin page."""
+    def GET_AUTH(self):  # pylint: disable=arguments-differ
+        """ Display admin page """
+        return self.show_page()
+
+    def POST_AUTH(self):  # pylint: disable=arguments-differ
+        """ Display admin page """
+        return self.show_page()
+
+    def show_page(self):
+        """Display page"""
+        return self.template_helper.render("admin/admin_page.html")
+
+
+class AdministrationUsersPage(INGIniousAdministratorPage):
+    """User Admin page."""
+    def GET_AUTH(self):  # pylint: disable=arguments-differ
+        """ Display admin users page """
+        return self.show_page()
+
+    def POST_AUTH(self):  # pylint: disable=arguments-differ
+        """ Display admin users page """
+        return self.show_page()
+
+    def show_page(self):
+        """Display page"""
+        all_users = self.user_manager.get_users()
+        return self.template_helper.render("admin/admin_users.html", all_users=all_users)
+
+
+class AdministrationUserActionPage(INGIniousAdministratorPage):
+    """Action on User Admin page."""
```

Comment:
There should be no need to define the POST methods if the classes inherit from ``INGIniousPage``.

Reply:
in fact i have to do that cause of the ajax requests. Am i able to set method argument of an ajax request to "POST_AUTH"?
In case of no redefinition those request lost there way.

Reply:
That's probably because the session is lost. The ajax request should send the cookies normally, as this is done for other such requests in the code (submission progress polling, ...). You should check what's going on when the request is processed by the ``INGIniousAuthPage.POST``method.

Remark 2. implies that anyone can perform the requests, which is a major security and privacy flaw.